tiledb/vfs.py

```diff
@@ -495,3 +496,28 @@ def write(self, buff: bytes):
         self._nbytes += nbytes
         self._offset += nbytes
         return nbytes
+
+    def readinto(self, buff: bytes) -> int:
+        """
+        Read bytes into a pre-allocated, writable bytes-like object b, and return the number of bytes read.
+
+        :param buff bytes:
+        """
+        size = len(buff)
+        if not self.readable():
+            raise IOError("Cannot read from write-only FileIO handle")
+        if self.closed:
+            raise IOError("Cannot read from closed FileIO handle")
+
```

Comment:
Technically this is also [supposed to fail](https://github.com/python/cpython/blob/ab8a5beb5faf23795ede1e6304aebbcf2e20e0aa/Lib/test/test_file.py#L57-L59) unless the mode is binary, but I don't think it is possible to open a VFS without the `b` flag.

Reply:
Doesn't `readable()` check that its `rb` mode?